homeassistant/components/paperless_ngx/__init__.py

```diff
@@ -1,20 +1,67 @@
 """The Paperless-ngx integration."""
 
-from homeassistant.const import Platform
+from datetime import timedelta
+
+from pypaperless import Paperless
+from pypaperless.exceptions import (
+    InitializationError,
+    PaperlessConnectionError,
+    PaperlessForbiddenError,
+    PaperlessInactiveOrDeletedError,
+    PaperlessInvalidTokenError,
+)
+
+from homeassistant.const import CONF_API_KEY, CONF_URL, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, LOGGER
+from .coordinator import (
+    PaperlessConfigEntry,
+    PaperlessData,
+    PaperlessStatisticCoordinator,
+    PaperlessStatusCoordinator,
+)
 
-from .coordinator import PaperlessConfigEntry, PaperlessCoordinator
+UPDATE_INTERVAL_STATISTICS = timedelta(seconds=120)
+UPDATE_INTERNAL_STATUS = timedelta(seconds=300)
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: PaperlessConfigEntry) -> bool:
     """Set up Paperless-ngx from a config entry."""
 
-    coordinator = PaperlessCoordinator(hass, entry)
-    await coordinator.async_config_entry_first_refresh()
+    api = await _get_paperless_api(hass, entry)
+
+    statistics_coordinator = PaperlessStatisticCoordinator(
+        hass, entry, api, "statistics", UPDATE_INTERVAL_STATISTICS
+    )
+    await statistics_coordinator.async_config_entry_first_refresh()
+
+    try:
+        status_coordinator = PaperlessStatusCoordinator(
+            hass, entry, api, "status", UPDATE_INTERNAL_STATUS
+        )
```

Comment:
why do we input both the string and the update interval? They never change so they can be declared in the coordinator.py instead